Client/Frontend/UserContent/UserScripts/Scripts_Dynamic/ScriptHandlers/Paged/SolanaProviderScriptHandler.swift

```diff
@@ -0,0 +1,346 @@
+  /// will encoded the response as a json object for success or provide an error dictionary for failures
+  @MainActor func signAllTransactions(args: String?) async -> (Any?, String?) {
+    guard let args = args,
+          let transactions = MojoBase.Value(jsonString: args)?.listValue,
+          let provider = tab?.walletSolProvider else {
+      return (nil, buildErrorJson(status: .invalidParams, errorMessage: "Invalid args"))
+    }
+    let params: [BraveWallet.SolanaSignTransactionParam] = transactions.compactMap { tx in
+      guard let transaction = tx.dictionaryValue,
+            let serializedMessage = transaction[Keys.serializedMessage.rawValue],
+            let signatures = transaction[Keys.signatures.rawValue] else {
+        return nil
+      }
+      return createSignTransactionParam(serializedMessage: serializedMessage, signatures: signatures)
+    }
+    guard !params.isEmpty else {
+      return (nil, buildErrorJson(status: .invalidParams, errorMessage: "Invalid args"))
+    }
+    let (status, errorMessage, serializedTxs) = await provider.signAllTransactions(params)
+    guard status == .success else {
+      return (nil, buildErrorJson(status: status, errorMessage: errorMessage))
+    }
+    guard let encodedSerializedTxs = JSONSerialization.jsObject(withNative: serializedTxs) else {
+      return (nil, buildErrorJson(status: .internalError, errorMessage: "Internal error"))
+    }
+    return (encodedSerializedTxs, nil)
+  }
+  
+  /// Helper function to build `SolanaSignTransactionParam` given the serializedMessage and signatures from the `solanaWeb3.Transaction`.
+  private func createSignTransactionParam(serializedMessage: MojoBase.Value, signatures: MojoBase.Value) -> BraveWallet.SolanaSignTransactionParam {
+    // get the serialized message
+    let serializedMessageValues = serializedMessage.bufferToList?.map { UInt8($0.intValue) } ?? []
+    let encodedSerializedMsg = (Data(serializedMessageValues) as NSData).base58EncodedString()
+    // get the signatures array
+    let signaturesValues = (signatures.listValue ?? []).compactMap { $0.dictionaryValue }
+    let signatures: [BraveWallet.SignaturePubkeyPair] = signaturesValues.map {
+      BraveWallet.SignaturePubkeyPair(
+        signature: $0[Keys.signature.rawValue]?.bufferToList?.map { NSNumber(value: $0.intValue) },
+        publicKey: $0[Keys.publicKey.rawValue]?.stringValue ?? ""
+      )
+    }
+    return .init(encodedSerializedMsg: encodedSerializedMsg, signatures: signatures)
+  }
+  
+  private func buildErrorJson(status: BraveWallet.SolanaProviderError, errorMessage: String) -> String? {
+    JSONSerialization.jsObject(withNative: [Keys.code.rawValue: status.rawValue, Keys.message.rawValue: errorMessage])
+  }
+  
+  @MainActor private func emitConnectEvent(publicKey: String) async {
+    if let webView = tab?.webView {
+      let script = "window.solana.emit('connect', new solanaWeb3.PublicKey('\(publicKey)'))"
```

Comment:
same question here whether public address sanitization would make sense here or not